File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.java
class Solution {
  public int[] runningSum(int[] nums) {
      for (int i = 1; i < nums.length; i++) {
          nums[i] += nums[i-1];
      }
      return nums;
  }
}

// Runtime: 0 ms (100.00%), Memory: 42.63 MB (31.64%)
// Time Complexity: O(n), where n is the number of elements in the array.
// Space Complexity: O(1), no extra space used.

File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        for i in range(1, len(nums)):
            nums[i] = nums[i-1] + nums[i]
        return nums

# Runtime: 0 ms (100.00%), Memory: 19.44 MB (24.72%)
# Time complexity: O(n)
# Space complexity: O(1)
